### DISCUSSION TITLE
Server side Redirects in 2020

### DISCUSSION THREAD

**Discussion by hdgone** [vercel/next.js#19401] [Help] [answered]
How do you guys do server side redirects right now?

Using this:
```javascript
// inside of getServerSideProps
if (context.res) {
    context.res.writeHead(303, { Location: target });
    context.res.end();
  }
```
works when you're typing the url (or you are not in the app) and going to the page but when you're already in the app and try to visit a page that has this logic implemented the whole app crashes:

`Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client`

When I'm trying to find a solution everyone says that "it's a known issue, use clientside redirects instead" or "here's an RFC" and these answers are kind of outdated. So I was wondering how do people implement SS redirects in their apps? I don't believe that everyone is just using client side redirects

**Comment by paambaati** [comment]
Server-side redirects have been supported since Next.js 9.5! See https://nextjs.org/blog/next-9-5#redirects

**Comment by rafaelalmeidatk** [comment, marked as answer]
Did you try the `redirect` option from `gSSP`? <https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering>

Here is the example from the docs:

```js
export async function getServerSideProps(context) {
  const res = await fetch(`https://.../data`)
  const data = await res.json()

  if (!data) {
    return {
      redirect: {
        destination: '/',
        permanent: false,
      },
    }
  }

  return {
    props: {}, // will be passed to the page component as props
  }
}
```

**Reply by hdgone** [reply]
This is brilliant! I've seen it in the RFC but thought it wasn't implemented yet. Also searching next.js docs didn't help here.
Works like a charm, thank you!
